shared/domains/chelonia/chelonia.js

```diff
@@ -62,18 +91,60 @@ sbp('sbp/selectors/register', {
       return stack
     }
   },
-  'chelonia/configure': function (config: ?Object) {
-    merge(this.config, config || {})
+  'chelonia/configure': function (config: Object) {
+    merge(this.config, config)
+    // merge will strip the hooks off of config.hooks when merging from the root of the object
+    // because they are functions and cloneDeep doesn't clone functions
+    merge(this.config.hooks, config.hooks || {})
+  },
+  'chelonia/connect': function (): Object {
+    if (!this.config.connectionURL) throw new Error('config.connectionURL missing')
+    if (!this.config.connectionOptions) throw new Error('config.connectionOptions missing')
+    if (this.socket) {
```

Comment:
Since `this.socket` here is actually not a WebSocket object, I think something like `this.client` or `this.pubsub` would be more correct